src/sentry/eventstream/snuba.py

```diff
@@ -161,6 +161,24 @@ def insert(
         event_type = self._get_event_type(event)
         occurrence_data = self._get_occurrence_data(event)
 
+        # instead of normalizing and doing custom 'contexts' processing in snuba, we elect to do it here instead to
+        # avoid having to clutter up snuba code with business logic
+        if event_type == EventStreamEventType.Generic:
+            if event_data.get("contexts"):
+                event_data = dict(event_data)
+                contexts = event_data.setdefault("contexts", {})
+
+                # add user.geo to contexts if it exists
+                user_dict = event_data.get("user", event_data.get("sentry.interfaces.User", {}))
+                geo = user_dict.get("geo", {})
```

Comment:
can we use `get_path()` here? this logic is a little dense

Reply:
Used this:
`event_data.get("user") or event_data.get("sentry.interfaces.User") or {}` which I think may be clearer.